### PR TITLE
[IOS-556]: Get rid of Awesome cache

### PR DESCRIPTION
**Pull request by yuliaveres**
Awesome cache caused us lots of inconvenience due to all disk operations being executed on main thread. (Already replaced on AvatarManager)

### CODE REVIEW

File: Toshi/APIClients/EthereumAPIClient.swift [hunk 1]
```diff
@@ -203,7 +200,7 @@ final class EthereumAPIClient {
             }
 
             DispatchQueue.main.async {
-                self?.cache.setObject(balance, forKey: EthereumAPIClient.CachedBalanceKey)
+                self?.cache.set(value: balance.stringValue, key: EthereumAPIClient.CachedBalanceKey)
```
Comment (designatednerd) [inline Toshi/APIClients/EthereumAPIClient.swift]: should this still be on the main thread?

File: Toshi/APIClients/idAPIClient.swift [hunk 2]
```diff
@@ -41,22 +40,6 @@ final class IDAPIClient: CacheExpiryDefault {
...
-    }()
-
-    private lazy var contactCache: Cache<TokenUser> = {
-        do {
-            return try Cache<TokenUser>(name: "tokenContactCache")
-        } catch {
-            fatalError("Couldn't instantiate the contact cache")
-        }
-    }()
-
```
Comment (designatednerd) [inline Toshi/APIClients/idAPIClient.swift]: why remove caching entirely from this  API?

File: Toshi/Models/Currency.swift [hunk 3]
```diff
@@ -38,153 +36,148 @@ final class CurrenciesCacheData: NSObject, NSCoding {
     }
 }
 
-final class Currency: NSObject, NSCoding {
+final class Currency: Codable {
```
Comment (designatednerd) [inline Toshi/Models/Currency.swift]: 🎉 `Codable` 🎉

File: Toshi/APIClients/EthereumAPIClient.swift (hunk 1 above)
Comment (yuliaveres) [reply Toshi/APIClients/EthereumAPIClient.swift]: not anymore

File: Toshi/APIClients/idAPIClient.swift (hunk 2 above)
Comment (yuliaveres) [reply Toshi/APIClients/idAPIClient.swift]: it is not currently used at all. (maybe was for searching previously ? no usage of any of these right now)

**Pull request merged**